### PR TITLE
Add an special object parameter to handle object creation in extension

### PR DESCRIPTION
**Pull request by D8H**
* It allow extension users to apply actions on created objects.

Project with a modified version of the "fire bullet" extension
- [CreatedObjectParameterTest.zip](https://github.com/4ian/GDevelop/files/11248257/CreatedObjectParameterTest.zip)

### CODE REVIEW

File: newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/ValueTypeEditor.js [hunk 1]
```diff
@@ -114,6 +114,12 @@ export default function ValueTypeEditor({
                 />
                 <SelectOption value="identifier" label={t`Identifier (text)`} />
                 <SelectOption value="scenevar" label={t`Scene variable`} />
+                {!isExpressionType && (
+                  <SelectOption
+                    value="objectListOrEmptyIfJustDeclared"
+                    label={t`Created objects`}
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/ValueTypeEditor.js:120]: Oh nice, very cool naming.

**Review by 4ian — APPROVED** [review]
Do you think we should add " (Advanced)" to show it's not usual to use this parameter?

Do you think we need more code generation integration tests about this to be confident this will work across complex situations?
My idea: 
Have a function that calls another function that create an object. Verify the picked object is properly sent back to the 2 parent context (the intermediate function and the top level caller).

**Comment by D8H** [comment]
> Do you think we should add " (Advanced)" to show it's not usual to use this parameter?

I think it's not necessary.

> Do you think we need more code generation integration tests about this to be confident this will work across complex situations? My idea: Have a function that calls another function that create an object. Verify the picked object is properly sent back to the 2 parent context (the intermediate function and the top level caller).

It's what the "fire bullet" extension does. I can try to cover it.

**Comment by D8H** [comment]
The test needs a bit of refactor.
I think I'll merge this PR and add the test in another PR.

**Pull request merged**